.github/workflows/validate-pr-actions.yaml

```diff
@@ -0,0 +1,108 @@
+---
+name: Validate Github Actions
+
+on:
+  workflow_call:
+    inputs:
+      image:
+        type: string
+        required: false
+  pull_request:
+    paths:
+      - actions/**
+      - tests/test_resources/mpyl_config.yml
+      - tests/test_resources/repository/changed_files.json
+      - tests/projects/job/**
+
+# If multiple commits are pushed simultaneously, only build the last one.
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number }}
+  cancel-in-progress: true
+
+jobs:
+
+  health-check:
+    name: ▶️ Run actions/health-check
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4.2.1
+
+      - name: Point action.yaml to a new version
+        if: inputs.image != ''
+        env:
+          IMAGE: docker://${{ inputs.image }}
+        run: yq -i '.runs.image = strenv(IMAGE)' actions/health-check/action.yaml
+
+      - uses: ./actions/health-check
+        with:
+          config-path: tests/test_resources/mpyl_config.yml
+
+  lint-projects:
+    name: ▶️ Run actions/lint-projects
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4.2.1
+
+      - name: Point action.yaml to a new version
+        if: inputs.image != ''
+        env:
+          IMAGE: docker://${{ inputs.image }}
+        run: yq -i '.runs.image = strenv(IMAGE)' actions/lint-projects/action.yaml
+
+      - uses: ./actions/lint-projects
+        with:
+          config-path: tests/test_resources/mpyl_config.yml
+
+  run-plan:
+    name: ▶️ Run actions/run-plan
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4.2.1
+
+      - name: Point action.yaml to a new version
+        if: inputs.image != ''
+        env:
+          IMAGE: docker://${{ inputs.image }}
+        run: yq -i '.runs.image = strenv(IMAGE)' actions/run-plan/action.yaml
+
+      - uses: ./actions/run-plan
+        with:
+          environment: pull-request
+          deploy-target: PullRequest
+          version: a-random-version
+          changed-files-path: tests/test_resources/repository/changed_files.json
+          config-path: tests/test_resources/mpyl_config.yml
+
+  run-stage:
+    name: ▶️ Run actions/run-stage
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4.2.1
+
+      - name: Point action.yaml files to a new version
+        if: inputs.image != ''
+        env:
+          IMAGE: docker://${{ inputs.image }}
+        run: |-
+          yq -i '.runs.image = strenv(IMAGE)' actions/run-plan/action.yaml
+          yq -i '.runs.image = strenv(IMAGE)' actions/run-stage/action.yaml
+
+      # To run a stage we always need to create a run plan before
```

Comment:
Until then, why not just put the runplan and stage jobs into 1? Saves us some time & resources

Reply:
Because this way we can easily see exactly what actions are being tested:
<img width="342" alt="image" src="https://github.com/user-attachments/assets/5e8273dc-559d-45e4-b4e4-c47f89435f75">